### PR TITLE
Update column headers for various coach report CSV files

### LINKED ISSUE #6151: Add more helpful strings for CSV export columns
Follow up on https://github.com/learningequality/kolibri/pull/6147

Since the CSV export feature was added after string freeze, we didn't have total flexibility in structuring CSV exports that may require additional columns and thus additional user strings. This is to follow up on the specific reports that could use more specific strings for the CSV export headers.

For example, some report tables describe learner status as "X learners have not started" and "Completed by X of Y learners". In the CSV export, the single column that describes these learner tallies becomes multiple columns with quantities for each status, so specific strings for these columns could be helpful.

With the flexibility to add strings, it's the developer's discretion to add more specificity or to restructure exported columns for optimal detail.

### Context

Target: next 0.13.x patch

### PR DESCRIPTION
**Pull request by jonboiser**
### Summary

Tries to shore up various TODO comments in the CSV export code

1. For Class Lesson/Quiz list, change `Recipients (1)` to `Recipient type` and `Recipients (2)` to `Recipients`. In the `Recipient type` cell, change possible values to `Whole Class|Groups|Groups and individuals`. In the `Recipients` cell, attempt to use `Intl.ListFormat` to localize comma separated lists.
1. In the Difficult Questions CSV, change `All` header to `All Learners`
1. Fix a bug in the QuizLessonDetailsHEader

### Reviewer guidance
1. Make sure that the `recipients` mixed-in method is used properly in the two components that use it. Make sure no other components use it.
1. Check CSV outputs where they have changed.
1. Make sure there are no other TODO/tech-debt things that are missed

### References

Fixes #6151

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [x] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/assets/src/utils/i18n.js [hunk 1]
```diff
@@ -262,3 +262,13 @@ export function localeCompare(str1, str2) {
...
+
+// Wrapper to Intl.ListFormat
+export function formatList(array) {
+  if (Intl.ListFormat) {
+    const formatter = new Intl.ListFormat(currentLanguage, { style: 'short', type: 'unit' });
+    return formatter.format(array);
+  } else {
+    return array.join(', ');
+  }
+}
```
Comment (bjester) [inline kolibri/core/assets/src/utils/i18n.js:274]: Awesome!

**Review by bjester — CHANGES_REQUESTED** [review]
Looks good! I have one more potential report improvement. 

On the Quiz report that lists the learners and their quiz progress, it has a "Progress" column, which we have currently being exported, but it also shows the number of questions answered underneath if they've started it. So I think it would be nice to add another column, something like `Questions answered` or `Number of questions answered`, to capture that in the export since it's visible on the page.

**Comment by jonboiser** [comment]
I noted the missing "Questions answered" thing in #6642. I'll add a patch for just the ReportsQuizLearnerListPage and we can tackle the rest in 6642 after we do more digging for other improvements we can make.

**Comment by jonboiser** [comment]
@bjester I added three new columns to this report:

1. Answered questions
2. Correct questions
3. Total questions

```
"Name","Progress","Score","Answered questions","Correct questions","Total questions","Groups"		
"jb","Completed","10%","3","1","10","kjhkhjk"
```

I'm noticing the numbers are actually coming out as strings. Is there anyway to get the CSV to just save these as numbers?

**Comment by bjester** [comment]
@jonboiser I don't see anything in the CSV generator lib. I think it's okay that the numbers are quoted.

**Review by bjester — APPROVED** [review]
Nice!

**Pull request merged**